STM32Cube/Core/Src/my2c.c

```diff
@@ -0,0 +1,144 @@
+#include <stdbool.h>
+#include "my2c.h"
+
+#include "stm32h7xx_hal.h"
+#include "stm32h7xx_hal_i2c.h"
+
+#include "FreeRTOS.h"
+#include "semphr.h"
+
+SemaphoreHandle_t I2CBinarySemaphore;
```

Comment:
For maximum specificity this should be named something like `I2C4BinarySemaphore`